Formula/lua-language-server.rb

```diff
@@ -0,0 +1,110 @@
+class LuaLanguageServer < Formula
+  desc "Language Server for the Lua language"
+  homepage "https://github.com/sumneko/lua-language-server"
+  # pull from git tag to get submodules
+  url "https://github.com/sumneko/lua-language-server.git",
+      tag:      "2.5.2",
+      revision: "cb2042160865589b5534a6bf0b6c366ae4ab1d99"
+  license "MIT"
+  head "https://github.com/sumneko/lua-language-server.git", branch: "master"
+
+  depends_on "ninja" => :build
+
+  on_linux do
+    depends_on "gcc" # For C++17
+  end
+
+  fails_with gcc: 5
+
+  # Disable `filesystem.test_appdata_path`.
+  #
+  # This test expeccts the application to run under the user's ${HOME},
+  # which is not compatible with homebrew's build environment.
+  # See https://github.com/actboy168/bee.lua/issues/21
+  # Adapted from https://github.com/macports/macports-ports/tree/master/lua/lua-language-server
+  patch :DATA
+
+  def install
+    ENV.cxx11
+    system "git", "submodule", "update", "--init", "--recursive"
+    chdir "3rd/luamake" do
+      system "compile/install.sh"
+    end
+    system "3rd/luamake/luamake", "rebuild"
+    bindir = if OS.mac?
+      "bin/macOS"
+    else
+      "bin/Linux"
+    end
+    (libexec/bindir).mkpath
+    (libexec/bindir).install "#{bindir}/lua-language-server"
+    (libexec/bindir).install "#{bindir}/main.lua"
+    libexec.install "main.lua", "debugger.lua"
+    libexec.install "locale", "meta", "script"
+    bin.write_exec_script ("#{libexec}/#{bindir}/lua-language-server")
```

Comment:
```suggestion
    bin.write_exec_script libexec/"bindir/lua-language-server"
```

Reply:
Wouldn't that be a literal `bindir/lua-language-server`? (It _has_ to be `foo/bar/lua-language-server` -- there is a hard-coded `../../main.lua` in there...

Reply:
I can make this `bin/arch` on both platforms, of course; I just thought that if I already need the actual platform in the `bindir` to get the built binary, I might as well reuse it. (The less divergence from the "real" build environment, the less chance of weird side effects...)

Reply:
Oops, yes, sorry, I meant to write
```suggestion
    bin.write_exec_script libexec/bindir/"lua-language-server"
```

Reply:
It thought I tried that... Let me check again!

Reply:
Yep, that works; must have had the quote in the wrong place!